<launch>
    <arg name="use_detect"                      default="true"/>
    <arg name="use_rotate_90_counter_clockwise" default="true"/>
    <arg name="is_display_origin_color_image"   default="false"/>
    <arg name="is_display_origin_depth_image"   default="false"/>
    <arg name="is_display_resize_depth_image"   default="false"/>

    <arg name="topic_camera_info_sub"           default="/color/camera_info"/>
    <arg name="topic_color_image_sub"           default="/color/image_raw"/>
    <arg name="topic_depth_image_sub"           default="/depth/image_rect_raw"/>  
    <arg name="distance_field_detect"           default="2.0"/> 
    <arg name="distance_field_warning"          default="1.5"/> 
    <arg name="distance_field_dangerous"        default="1.0"/>
    <arg name="camera"                          default="/depth_camera"/>
    <arg name="number_dangerous_point"          default="100"/>
    <arg name="number_warning_point"            default="100"/>
    <arg name="number_detect_point"             default="100"/>
    <arg name="json_file_path"                  default="$(find detect_obstacle)/config/custom.json"/>

    <arg name="camera_left"                     default="/depth_camera_left"/>
    <arg name="usb_port_id_left"                default="2-4"/>
    <arg name="top_image_left_resize"           default="238"/>
    <arg name="bottom_image_left_resize"        default="238"/>
    <arg name="left_image_left_resize"          default="200"/>
    <arg name="right_image_left_resize"         default="200"/>

    <arg name="camera_right"                    default="/depth_camera_right"/>
    <arg name="usb_port_id_right"               default="2-3"/>
    <arg name="top_image_right_resize"          default="238"/>
    <arg name="bottom_image_right_resize"       default="238"/>
    <arg name="left_image_right_resize"         default="200"/>
    <arg name="right_image_right_resize"        default="200"/>

    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="camera"                      value="$(arg camera_left)"/>
        <arg name="usb_port_id"                 value="$(arg usb_port_id_left)"/>
        <arg name="json_file_path"              value="$(arg json_file_path)"/>
    </include>
    <node name="detect_obstacle_depth_camera" pkg="detect_obstacle" type="detect_obstacle_depth_camera.py" output="screen" ns="$(arg camera_left)">
        <param name="use_rotate_90_counter_clockwise"   value="$(arg use_rotate_90_counter_clockwise)" />
        <param name="is_display_origin_color_image"     value="$(arg is_display_origin_color_image)" />
        <param name="is_display_origin_depth_image"     value="$(arg is_display_origin_depth_image)" />
        <param name="is_display_resize_depth_image"     value="$(arg is_display_resize_depth_image)" />
        <param name="camera"                            value="$(arg camera_left)" />
        <param name="topic_camera_info_sub"             value="$(arg topic_camera_info_sub)" />
        <param name="topic_color_image_sub"             value="$(arg topic_color_image_sub)" />
        <param name="topic_depth_image_sub"             value="$(arg topic_depth_image_sub)" />
        <param name="top_image_resize"                  value="$(arg top_image_left_resize)" />
        <param name="bottom_image_resize"               value="$(arg bottom_image_left_resize)" />
        <param name="left_image_resize"                 value="$(arg left_image_left_resize)" />
        <param name="right_image_resize"                value="$(arg right_image_left_resize)" />
        <param name="distance_field_detect"             value="$(arg distance_field_detect)" />
        <param name="distance_field_warning"            value="$(arg distance_field_warning)" />
        <param name="distance_field_dangerous"          value="$(arg distance_field_dangerous)" />
        <param name="number_dangerous_point"            value="$(arg number_dangerous_point)" />
        <param name="number_warning_point"              value="$(arg number_warning_point)" />
        <param name="number_detect_point"               value="$(arg number_detect_point)" />
    </node>
    
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="camera"                      value="$(arg camera_right)"/>
        <arg name="usb_port_id"                 value="$(arg usb_port_id_right)"/>
        <arg name="json_file_path"              value="$(arg json_file_path)"/>
    </include>
    <node name="detect_obstacle_depth_camera" pkg="detect_obstacle" type="detect_obstacle_depth_camera.py" output="screen" ns="$(arg camera_right)">
        <param name="use_rotate_90_counter_clockwise"   value="$(arg use_rotate_90_counter_clockwise)" />
        <param name="is_display_origin_color_image"     value="$(arg is_display_origin_color_image)" />
        <param name="is_display_origin_depth_image"     value="$(arg is_display_origin_depth_image)" />
        <param name="is_display_resize_depth_image"     value="$(arg is_display_resize_depth_image)" />
        <param name="camera"                            value="$(arg camera_right)" />
        <param name="topic_camera_info_sub"             value="$(arg topic_camera_info_sub)" />
        <param name="topic_color_image_sub"             value="$(arg topic_color_image_sub)" />
        <param name="topic_depth_image_sub"             value="$(arg topic_depth_image_sub)" />
        <param name="top_image_resize"                  value="$(arg top_image_right_resize)" />
        <param name="bottom_image_resize"               value="$(arg bottom_image_right_resize)" />
        <param name="left_image_resize"                 value="$(arg left_image_right_resize)" />
        <param name="right_image_resize"                value="$(arg right_image_right_resize)" />
        <param name="distance_field_detect"             value="$(arg distance_field_detect)" />
        <param name="distance_field_warning"            value="$(arg distance_field_warning)" />
        <param name="distance_field_dangerous"          value="$(arg distance_field_dangerous)" />
        <param name="number_dangerous_point"            value="$(arg number_dangerous_point)" />
        <param name="number_warning_point"              value="$(arg number_warning_point)" />
        <param name="number_detect_point"               value="$(arg number_detect_point)" />
    </node>

    <node name="multiple_depth_camera" pkg="detect_obstacle" type="multiple_depth_camera.py" output="screen" ns="$(arg camera)">
        <param name="camera"                            value="$(arg camera)" />
        <param name="camera_left"                       value="$(arg camera_left)" />
        <param name="camera_right"                      value="$(arg camera_right)" />
    </node>

    <arg name="open_rviz" default="false"/>
    <group if="$(arg open_rviz)"> 
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find detect_obstacle)/rviz/depth_image.rviz" required="true" />
    </group>
</launch>